compiler/src/iree/compiler/Dialect/Flow/TransformExtensions/FlowExtensionsOps.td

```diff
@@ -180,33 +180,72 @@ def MovePrecedingOpIntoDispatchRegionOp : Op<
 
 def CloneSucceedingOpIntoDispatchRegionOp : Op<
     Transform_Dialect, "iree.clone_succeeding_op_into_dispatch_region", [
-      // TODO: this is missing effects.
+      MemoryEffectsOpInterface,
       TransformOpInterface]> {
   let description = [{
     Clone the `target` op into the given dispatch region op. The dispatch region
     handle must be mapped to exactly one payload op.
 
     All operands of the target are replaced with values that are defined inside
-    of the dispatch region when possible.
+    of the dispatch region when possible. No new results are added to the
+    dispatch region.
 
-    If `update_uses_outside_of_region` is set (default value: `true`), all uses
-    of the original target op are replaced: The results of the cloned target op
-    are yielded from the dispatch region and used instead of results of the
-    original target op.
+    #### Return modes
+
+    `dispatch_region` must be mapped to exactly one op. `target` may be mapped
+    to multiple ops.
+
+    This transform reads both the `target` handle and the `dispatch_region`
+    handle. It produces a new handle to the cloned ops.
+  }];
+
+  let arguments = (ins Arg<PDL_Operation, "",
+                           [TransformMappingRead,
+                            PayloadIRRead]>:$target,
+                       Arg<PDL_Operation, "",
+                           [TransformMappingRead,
+                            PayloadIRRead,
```

Comment:
IIRC we cannot use PayloadIRRead from Tablegen as it does not attach to the right things.
I had to implement a proper `::getEffects` recently (my memory is a bit blurry here)

@ftynse

Reply:
Indeed. Putting side effect descriptions inside `Arg` associates them with the particular operand. So this means the effect of PayloadIRRead is on operands and may potentially be reordered with PayloadIRWrites affecting different (non-aliasing) values. So we have to implement `getEffects` manually instead to make sure the PayloadIR effects are not associated with any specific value. Alternatively, we can update the ODS generator to support this case.